ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/context/KsqlRestServiceContextFactory.java

```diff
@@ -18,22 +18,48 @@
 import io.confluent.ksql.services.DefaultServiceContext;
 import io.confluent.ksql.services.ServiceContext;
 import io.confluent.ksql.util.KsqlConfig;
+
+import java.util.Objects;
+import java.util.Optional;
+import javax.inject.Inject;
+import javax.ws.rs.container.ContainerRequestContext;
+
+import org.apache.kafka.streams.KafkaClientSupplier;
+import org.apache.kafka.streams.processor.internals.DefaultKafkaClientSupplier;
 import org.glassfish.hk2.api.Factory;
 
 /**
  * This class implements {@link Factory}, which allows a REST application to create
  * a new {@link ServiceContext} during REST requests.
  */
 public class KsqlRestServiceContextFactory implements Factory<ServiceContext> {
-  private final KsqlConfig ksqlConfig;
+  private static KsqlConfig ksqlConfig;
+
+  public static void configure(final KsqlConfig ksqlConfig) {
```

Comment:
I think this should be syncrhonized

Reply:
even if it is a single variable assignment inside?

Reply:
maybe not necessary... but it's basically free :)

Reply:
there's only one thread that uses it right?

Reply:
Yes, one single thread. It is initialized from the `KsqlRestApplication` only once.